### PR TITLE
Add javascript utils functions

### PR DESCRIPTION
**Pull request by YannBcf**
wip & open to discussions
This pull request adds:

client:
- requestAnimDict
- requestAnimSet
- requestClipSet
- requestCutscene

### CODE REVIEW

File: shared/bindings/Utils.js [hunk 1]
```diff
@@ -5,8 +5,11 @@
 // Client only
 if(alt.isClient) {
     alt.Utils.requestModel = async function(model, timeout = 1000) {
+        if (typeof timeout !== "number") throw new Error("Expected a number as second argument");
+
         if (typeof model === "string") model = alt.hash(model);
         if (!native.isModelValid(model)) throw new Error("Model not valid");
```
Comment (Stuyk) [inline shared/bindings/Utils.js:12]: Error should give more information other than `model not valid`.

Example: `Model 'xyz', with hash 'xyz' is invalid`

File: shared/bindings/Utils.js [hunk 2]
```diff
@@ -5,8 +5,11 @@
 // Client only
 if(alt.isClient) {
     alt.Utils.requestModel = async function(model, timeout = 1000) {
+        if (typeof timeout !== "number") throw new Error("Expected a number as second argument");
```
Comment (Stuyk) [inline shared/bindings/Utils.js:9]: Check for `typeof timeout === "string"` and try to parse into a number as well.

File: shared/bindings/Utils.js [hunk 3]
```diff
@@ -15,6 +18,56 @@ if(alt.isClient) {
             await alt.Utils.wait(5);
         }
     }
+
+    alt.Utils.requestAnimDict = async function(animDict) {
+        if (typeof animDict !== "string") throw new Error("Expected a string as first argument");
+        if (!native.doesAnimDictExist(animDict)) throw new Error("Anim dict not valid");
```
Comment (Stuyk) [inline shared/bindings/Utils.js:27]: What `anim dict` is invalid?

File: shared/bindings/Utils.js [hunk 4]
```diff
@@ -15,6 +18,56 @@ if(alt.isClient) {
...
+    }
+
+    alt.Utils.requestAnimSet = async function(animSet) {
+        if (typeof animSet !== "string") throw new Error("Expected a string as first argument");
+
+        const checkUntil = Date.now() + timeout;
+        native.requestAnimSet(animSet);
+
+        while (!native.hasAnimSetLoaded(animSet)) {
+            if (Date.now() > checkUntil) throw new Error("Failed to request anim set");
```
Comment (Stuyk) [inline shared/bindings/Utils.js:46]: What `anim set` failed to request?

File: shared/bindings/Utils.js [hunk 5]
```diff
@@ -15,6 +18,56 @@ if(alt.isClient) {
...
+    }
+
+    alt.Utils.requestClipSet = async function(clipSet) {
+        if (typeof clipSet !== "string") throw new Error("Expected a string as first argument");
+
+        const checkUntil = Date.now() + timeout;
+        native.requestClipSet(clipSet);
+
+        while (!native.hasClipSetLoaded(clipSet)) {
+            if (Date.now() > checkUntil) throw new Error("Failed to request clip set");
```
Comment (Stuyk) [inline shared/bindings/Utils.js:59]: What `clip set` failed to request?

File: shared/bindings/Utils.js [hunk 6]
```diff
@@ -15,6 +18,56 @@ if(alt.isClient) {
...
+            await alt.Utils.wait(5);
+        }
+    }
+
+    alt.Utils.requestCutscene = async function(cutsceneName, flags) {
+        if (typeof cutsceneName !== "string") throw new Error("Expected a string as first argument");
+        if (typeof flags !== "number") throw new Error("Expected a number as second argument");
```
Comment (Stuyk) [inline shared/bindings/Utils.js:66]: Second argument should try to parse a string as well.

File: shared/bindings/Utils.js [hunk 7]
```diff
@@ -15,6 +18,56 @@ if(alt.isClient) {
...
+
+        const checkUntil = Date.now() + timeout;
+        native.requestCutscene(cutsceneName, flags);
+
+        while (!native.hasCutsceneLoaded(cutsceneName)) {
+            if (Date.now() > checkUntil) throw new Error("Failed to request cutscene");
```
Comment (Stuyk) [inline shared/bindings/Utils.js:73]: What `cutscene` failed to request?

File: shared/bindings/Utils.js [hunk 8]
```diff
@@ -23,7 +76,22 @@ else {
...
+    if (typeof timeout !== "number") throw new Error("Expected a number as second argument");
+
     return new Promise(resolve => {
         alt.setTimeout(resolve, timeout);
     });
 }
+
+alt.Utils.timeoutPromise = async function(promise, timeout) {
+    if (!(promise instanceof Promise)) throw new Error("Expected a promise as first argument");
+    if (typeof timeout !== "number") throw new Error("Expected a number as second argument");
```
Comment (Stuyk) [inline shared/bindings/Utils.js]: Try parsing string as well.

**Review by Stuyk — CHANGES_REQUESTED** [review]

File: shared/bindings/Utils.js (hunk 2 above)
Comment (LeonMrBonnie) [reply shared/bindings/Utils.js:9]: Let's not get overboard with these type conversions, for a model you may be reading it out of a database and then get it as a string or something, but a timeout like that will almost certainly be hardcoded into your code, so specifying that as a string there makes zero sense.

File: shared/bindings/Utils.js [hunk 9]
```diff
@@ -23,7 +76,22 @@ else {
 
 // Shared
 alt.Utils.wait = function(timeout) {
+    if (typeof timeout !== "number") throw new Error("Expected a number as second argument");
```
[the remark this answers is not in the document]
Comment (LeonMrBonnie) [reply shared/bindings/Utils.js]: See comment above

**Review by LeonMrBonnie — APPROVED** [review]

**Pull request merged**